ui/src/pages/backoffice/Loan/LoanSearch/SearchDateRange.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react';
+import { Card } from 'semantic-ui-react';
+import { DatePicker } from '@components';
+import { withState } from 'react-searchkit';
+import _isEmpty from 'lodash/isEmpty';
+
+class _SearchDateRange extends Component {
+  getCurrentDates() {
+    const { filters } = this.props.currentQueryState;
+    let fromDate = '';
+    let toDate = '';
+
+    filters.forEach(([name, value]) => {
+      if (name === 'loans_from_date') fromDate = value;
+      if (name === 'loans_to_date') toDate = value;
+    });
+    return [fromDate, toDate];
+  }
+
+  /** react-searchkit allows having the same filter multiple times.
+   * For the range dates filters we want each filter one time only so we have
+   * to remove any pre-existing filters with the same name
+   */
+  onDateChange = newFilter => {
+    const [name, value] = newFilter;
+    let filters = newFilter;
+    // If value is empty we simply remove the filter otherwise if we have
+    // value we remove the filter and add the new one.
+    if (!_isEmpty(value)) filters = [[name, ''], newFilter];
```

Comment:
To remove the previously selected filter, shouldn't you pass the already selected value first?
E.g. currently I have `loan_start_date:2020-01-01` if I select the date `loan_start_date:2019-01-01`, shouldn't we pass:
`[['loan_start_date', '2020-01-01'], ['loan_start_date', '2019-01-01']]`

Reply:
That was indeed my initial understanding, but in the [searchkit](https://github.com/inveniosoftware/react-searchkit/blob/b74306908169cd226cfd3d35a4f95aa8bc67921b/src/lib/state/selectors/query.js#L71) the comparison we do not check for the value, we only check if it matches the starting part.

Reply:
Unfortunately this is not the case, see tests that you wrote too. This is related to [this bug](https://github.com/inveniosoftware/react-searchkit/issues/85): it happens that since the string is empty, it clears out all filters selected for this filter. If we fix that bug, we might break this code.
Please change it to something like (pseudo-code):
```
onDateChange = newSelectedValue => {
    const resetPreviousValue = getFromRSKCurrentQueryState()
    const newFilters = [resetPreviousValue];
    if (!isEmpty(newSelectedValue)) {
        newFilters.push(newSelectedValue);
    }
    return this.props.updateQueryState({ filters: newFilters });
  };
```